File: backend/app/schemas/viewer.py
from datetime import datetime

from pydantic import BaseModel


class ViewerResponse(BaseModel):

    id: int

    original_filename: str

    filename: str

    file_type: str

    file_path: str

    status: str

    upload_date: datetime

    language: str

    confidence: float

    processing_time: float

    text: str

    class Config:
        from_attributes = True

File: frontend/src/api/viewer.ts
import api from "@/api/axios";

export interface ViewerDocument {

    id: number;

    original_filename: string;

    filename: string;

    file_type: string;

    file_path: string;

    status: string;

    upload_date: string;

    language: string;

    confidence: number;

    processing_time: number;

    text: string;

}

export const ViewerAPI = {

    async getDocument(id: number): Promise<ViewerDocument> {

        const response = await api.get<ViewerDocument>(
            `/viewer/${id}`
        );

        return response.data;

    }

};